test/ninja/ninja_test_sconscripts/sconstruct_ninja_determinism

```diff
@@ -0,0 +1,20 @@
+import SCons
+import random
+SetOption('experimental','ninja')
+SetOption('skip_ninja_regen', True)
```

Comment:
If I comment this line out, the test still passes? Is that expected?

Reply:
yeah actually this flag is not relevant to the current test. This PR makes ninja files deterministic, but the flag is for controlling skipping the regeneration and restarting of the daemon. The flag is to protect users who may be using Glob or SCons generated files in command lines. Its assumed if the user enables the flag, they know what they are doing.

I will update this test to also make sure the mtime of the ninja file did not change, which would indicate it was never overwritten.

Reply:
o.k. so run this test with master branch code and it should fail then?

Reply:
yeah I test that